src/basic/errno-list.h

```diff
@@ -2,14 +2,25 @@
 #pragma once
 
 #include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
 /*
  * MAX_ERRNO is defined as 4095 in linux/err.h
  * We use the same value here.
  */
 #define ERRNO_MAX 4095
 
+#if HAVE_STRERRORNAME_NP
+static inline const char* errno_to_name(int id) {
+        return strerrorname_np(abs(id));
```

Comment:
The man pages says if the input is zero, then `strerrorname_np()` provides a string `"0"`. But our implement provides `NULL`. I think it is better to make the behavior consistent.
Also, `errno_from_name()` does not return `0` when the input is string `"0"`. To make _to_name() and _from_name() consistent, I think it is better that this returns `NULL` when `id` is zero.